src/main/java/at/ac/tuwien/kr/alpha/common/rule/head/impl/NormalHead.java

```diff
@@ -0,0 +1,66 @@
+package at.ac.tuwien.kr.alpha.common.rule.head.impl;
+
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.rule.head.Head;
+
+/**
+ * Copyright (c) 2019, the Alpha Team.
+ */
+public class NormalHead extends Head {
+
+	private final Atom atom;
+
+	public NormalHead(Atom atom) {
+		this.atom = atom;
+	}
+
+	// TODO ideally this should be inherited from Head, but need to check how to handle choiceHeads
```

Comment:
Should this TODO be addressed before the pull request is accepted?

Reply:
Not really - I put that TODO there as a reminder to all of us to think about it, but can't come up with a good answer right now... On one hand, we could put ```isGround``` as an abstract method into ```Head``` and implement it for ```ChoiceHead``` such that a ```ChoiceHead``` is ground iff the ```lowerElement```, ```upperElement``` and all ```ChoiceElement```s are ground, but on the other hand, it would be dead code anyway since ```ChoiceHead```s are rewritten before solving, so it'd never be called in the current workflow. I'd like to keep the TODO there, to be taken care of once someone implements support for bounded choices, maybe that would be something during which that decision should be made.

Reply:
Thanks for the feedback! I transformed the TODO to a note.